src/main/java/org/ngafid/uploads/process/format/ParquetFileProcessor.java

```diff
@@ -154,27 +112,43 @@ private void extractColumns(GenericRecord record,
                 continue;
             }
 
-            // Collect values for each column
-            // Parquet files have time values in milliseconds, convert them to seconds.
-            String rawTime = getStringOrNaN(element, "time");
-
             try {
-                double timeValue = Double.parseDouble(rawTime);
-                double timeInSeconds = (timeValue > 32503680000L) ? timeValue / 1000 : timeValue;
+                double unixTime = parseSafeDouble(getStringOrNaN(element, "time"));
+                double timeInSeconds = (unixTime > 32503680000L) ? unixTime / 1000 : unixTime;
                 timeValues.add(timeInSeconds);
+                try {
+                    utcDateTimes.add(TimeUtils.convertUnixTimeToUTCDateTime(timeInSeconds));
+                } catch (NumberFormatException e) {
+                    utcDateTimes.add("Error");
+                }
             } catch (NumberFormatException e) {
-                LOG.warning("Invalid time format, setting as NaN: " + rawTime);
-
+                timeValues.add(Double.NaN);
+                utcDateTimes.add("Error");
```

Comment:
IS this something you encountered frequently during development? I fear what an empty entry in the UTC column might do. If the first few entries are empty we can just skip them until the data begins, but adding an invalid date to the date series seems potentially hazardous.

IF you didn't encounter this during development, I'd recommend just propagating the error upwards which should effectively declare the file to be invalid.